src/addrdb.h

```diff
@@ -16,12 +17,11 @@ class CSubNet;
 class CAddrMan;
 class CDataStream;
 
-typedef enum BanReason
-{
-    BanReasonUnknown          = 0,
-    BanReasonNodeMisbehaving  = 1,
-    BanReasonManuallyAdded    = 2
-} BanReason;
+enum BanReason {
+    BanReasonUnknown,
+    BanReasonNodeMisbehaving,
+    BanReasonManuallyAdded
+};
```

Comment:
This is just asking to be a `enum class`.

Reply:
It is not trivial as it involves (de)serialization, and requires some `static_cast`s. Could we leave it out of this PR scope?

Reply:
Please drop c9862883713ca3ee008444e132d961266331397f then, it's unrelated to the PR goal and it's already generating unnecessary "noise".

Reply:
Yes, please make it fully move-only then.

Reply:
Agree.